### PR TITLE
Feature: add form from config

### PR DESCRIPTION
**Pull request by kseniaSs**
Added:

- Create form from user configuration
- Send real data to notion from form
- Collapse form after sending

We have an example configuration in configuration.ts.

**For working:**

1) You should add 'product_type' && 'product_size' && 'software_license' && 'suggestions' fields to your Notion's database. Without these fields it's not working.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -6,11 +6,18 @@ import { exampleConfiguration } from './configuration';
...
  */
 function createWidget(): void {
-  const form = new Form();
-  const notion = new Notion(exampleConfiguration);
+  const widget = new Form(exampleConfiguration.form);
+  const notion = new Notion(exampleConfiguration.notion);
 
-  form.container.addEventListener('click', () => {
-    notion.send('Hello');
+  widget.form.addEventListener('submit', (event) => {
```
Comment (TatianaFomina) [inline src/app.ts:12]: maybe make `form` a  private field of Form class and just pass onSubmit callback to constructor?

File: src/form.ts [hunk 2]
```diff
@@ -30,9 +66,49 @@ export class Form {
       textContent: 'Take a 2-minutes survey🙏',
```
Comment (TatianaFomina) [inline src/form.ts:66]: don't we want to allow to set these values via configuration as well?

File: src/form.ts [hunk 3]
```diff
@@ -30,9 +66,49 @@ export class Form {
...
+      } else {
+        this.openWidget();
+      }
+    });
+  }
+
+  /**
+   * Create and add form to document
+   *
+   * @returns {HTMLFormElement}
```
Comment (TatianaFomina) [inline src/form.ts:87]: no need to add `@returns` to jsdoc as it's typescript and return type is already specified

File: src/styles/form.module.css [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  padding: 10px;
+  border: 1px solid color-border;
+  box-shadow: 0 7px 9px color-box-shadow;
+  border-radius: 10px;
+  font-family: system-ui;
+  bottom: 10px;
+  right: 10px;
+  position: fixed;
+  z-index: 2;
+  /* z-index: 2147483647; */
```
Comment (TatianaFomina) [inline src/styles/form.module.css:16]: can be removed?

File: src/styles/form.module.css [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  font-weight: bold-font-weight;
+}
+
+.select {
+  height: 32px;
+  padding: 0 8px;
+  border-radius: 6px;
+  border: 1px solid color-select-border;
+  background-repeat: no-repeat;
+  background-image: url("data:image/svg+xml,%3Csvg width='10' height='7' viewBox='0 0 10 7' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M0.833374 1.33333L4.85862 5.35858C4.93672 5.43668 5.06336 5.43668 5.14146 5.35858L9.16671 1.33333' stroke='black' stroke-width='1.66667' stroke-linecap='round'/%3E%3C/svg%3E%0A");
```
Comment (TatianaFomina) [inline src/styles/form.module.css:46]: let's use icon from [codex-team/icons](https://github.com/codex-team/icons)

**Comment by TatianaFomina** [comment]
Wrong font in textarea

telegram-cloud-photo-size-2-5271899595531732020-x

File: src/configuration.ts [hunk 6]
```diff
@@ -1,15 +1,115 @@
...
+          type: TypeField.Textarea,
+          placeholder: 'Enter your ideas, requests or issues',
+        },
+      },
+      {
+        field: {
+          type: TypeField.Submit,
+          textContent: 'Submit',
+        },
+      },
```
Comment (neSpecc) [inline src/configuration.ts:112]: I think the Submit shouldn't be in a config. I'll be always added to the end of the form.

File: src/service/notion/index.ts [hunk 7]
```diff
@@ -29,29 +29,36 @@ export class Notion implements Service {
...
+            rich_text: [
+              {
+                type: 'text',
+                text: { content: content[key] },
+              },
+            ],
+          },
+        }),
+      {}
+    );
```
Comment (neSpecc) [inline src/service/notion/index.ts:56]: this could be moved to the separate method I guess

**Review by TatianaFomina — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Review by ilyamore88 — APPROVED** [review]

**Pull request merged**